pkg/vm/vm.go

```diff
@@ -569,14 +569,26 @@ func (v *VM) execute(ctx *Context, op Instruction) {
 
 	// Object operations.
 	case NEWARRAY:
-		n := v.estack.Pop().BigInt().Int64()
-		items := make([]StackItem, n)
-		v.estack.PushVal(&ArrayItem{items})
+		item := v.estack.Pop()
+		switch t := item.value.(type) {
```

Comment:
Also notice that the C# code actually makes `NEWARRAY` applied to array and `NEWSTRUCT` applied to struct a noop, and this was done intentionally, so we need to support this behavior (and even better test for it).

Reply:
`Yes, I have missed this. Updated PR.